lib/galaxy/jobs/runners/util/job_script/DEFAULT_JOB_FILE_TEMPLATE.sh

```diff
@@ -4,6 +4,10 @@
 # to ensure this script is runnable before running it directly
 # or submitting it to a cluster manager.
 if [ -n "$ABC_TEST_JOB_SCRIPT_INTEGRITY_XYZ" ]; then
+    $command
```

Comment:
Good point, we should directly test the tool script. 

I'm not a huge fan of propagating hacks, but haven't spent any time coming up with a better solution -- imho, it is likely the time waiting/sleeps that are having an affect (I see the error on AFS), not necessarily the non-blocking sync command (still get spurious failures after 10 loops at .25s -- locally, I've arbitrarily set it to 1000).

If you are volunteering to hack up a better fix, that would be great :beers: